sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -104,6 +105,31 @@ abstract class JdbcDialect extends Serializable with Logging {
    */
   def getJDBCType(dt: DataType): Option[JdbcType] = None
 
+  /**
+   * Convert java.sql.Timestamp to a LocalDateTime representing the same wall-clock time as the
+   * value stored in a remote database.
+   * JDBC dialects should override this function to provide implementations that suite their
+   * JDBC drivers.
+   * @param t Timestamp returned from JDBC driver getTimestamp method.
+   * @return A LocalDateTime representing the same wall clock time as the timestamp in database.
+   */
+  @Since("3.5.0")
+  def convertJavaTimestampToTimestampNTZ(t: Timestamp): LocalDateTime = {
```

Comment:
they are both timestamps in Java. How about `def toLocalDateTime` and `def toSQLTimestamp`?

Reply:
I would prefer to keep the names as they are.

`toLocalDateTime` would collide with `Timestamp.toLocalDateTime` method which already exists. Also, I would like to emphasis that this is not a general Java type to Java type conversion, but special cased under the use case/interpretation of Spark TimestampNTZType.

Reply:
SGTM